Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift

```diff
@@ -1,20 +1,244 @@
+            .compactMap({$0})
+            .sink( receiveValue: { image in
+                DispatchQueue.main.async {
+                    self.processingState = .confirmation(image)
+                }
+            })
+    }
+
+    func captureImage() {
+        cameraManager.capturePhoto()
     }
 
     func resetState() {
 
     }
 
+    func cropImage(_ capturedImage: UIImage, quadView: QuadrilateralView) {
+        guard let quad = quadView.quad, let ciImage = CIImage(image: capturedImage) else {
+            return
+        }
+        let cgOrientation = CGImagePropertyOrientation(capturedImage.imageOrientation)
+        let orientedImage = ciImage.oriented(forExifOrientation: Int32(cgOrientation.rawValue))
+        let scaledQuad = quad.scale(quadView.bounds.size, capturedImage.size)
+
+        // Cropped Image
+        var cartesianScaledQuad = scaledQuad.toCartesian(withHeight: capturedImage.size.height)
+        cartesianScaledQuad.reorganize()
+
+        let filteredImage = orientedImage.applyingFilter("CIPerspectiveCorrection", parameters: [
+            "inputTopLeft": CIVector(cgPoint: cartesianScaledQuad.bottomLeft),
+            "inputTopRight": CIVector(cgPoint: cartesianScaledQuad.bottomRight),
+            "inputBottomLeft": CIVector(cgPoint: cartesianScaledQuad.topLeft),
+            "inputBottomRight": CIVector(cgPoint: cartesianScaledQuad.topRight)
+        ])
+
+        frontImage = UIImage.from(ciImage: filteredImage)
+    }
+
+    func resumeCameraSession() {
+        cameraManager.resumeSession()
+    }
+
     func pauseCameraSession() {
+        cameraManager.pauseSession()
+    }
 
+    func handleCompletion() {
+        switch processingState {
+        case .complete(let response, let error):
+            pauseCameraSession()
+            processingState = .endFlow
+            if let error = error {
+                captureResultDelegate?.didError(error: error)
+                return
+            }
+            captureResultDelegate?.didSucceed(documentFrontImage: Data(),
+                                              documentBackImage: nil,
+                                              jobStatusResponse: response)
+        default:
+            break
+        }
+    }
+
+    func submit(navigation: NavigationViewModel) {
+        if captureBothSides && postion == .front {
+            processingState = nil
+            postion = .back
+            navigation.navigate(destination: .documentBackCaptureInstructionScreen(documentCaptureViewModel: self,
+                                                                                             delegate: captureResultDelegate),
+                                         style: .push)
+        } else {
+            processingState = .inProgress
+            navigation.navigate(destination: .selfieCaptureScreen(selfieCaptureViewModel: SelfieCaptureViewModel(userId: userId, jobId: jobId, isEnroll: false), delegate: nil), style: .push)
```

Comment:
We should be setting a delegate for this and handling result or error as appropriate